src/Microsoft.Diagnostics.Monitoring.WebApi/ICollectionRuleService.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal interface ICollectionRuleService : IHostedService, IAsyncDisposable
+    {
+        public Task ApplyRules(
+            IEndpointInfo endpointInfo,
+            CancellationToken token);
+
+        public Task RemoveRules(
+            IEndpointInfo endpointInfo,
+            CancellationToken token);
+
+        public Dictionary<string, CollectionRuleDescription> GetCollectionRulesDescriptions(IEndpointInfo endpointInfo);
```

Comment:
consider IDictionary

Reply:
I ended up getting an error when doing that ("cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type"); I noticed for `GetProcessEnvironment` we're just using `Dictionary`, so I left this as-is for now.